### ISSUE TITLE
[Bug]: unable to build using new architecture

### ISSUE THREAD

**Issue by ottob** [braze-inc/braze-react-native-sdk#250]
### Which Platforms?

iOS

### Which React Native Version?

0.73.4

### Which @braze/react-native-sdk SDK version?

8.3.0

### Repro Rate

100%

### Steps To Reproduce

Im building using expo EAS. I have @braze/expo-plugin v2.0.0 and expo 50 sdk

Using VM template "macos-ventura-13.6-xcode-15.2" running macOS Ventura 13.6
Installed software:
- Xcode 15.2 (15C500b)
- Node.js 18.18.0

using an app.config.js with:
```
'expo-build-properties',
        {
          android: {
            newArchEnabled: true,
          },
          ios: {
            useFrameworks: 'static', // we need this for react native firebase
            newArchEnabled: true,
          },
        },
```

### Expected Behavior

Successful build

### Actual Incorrect Behavior

The build failed with:
```
/Users/expo/workingdir/build/apps/patient-native/ios/build/generated/ios/BrazeReactModuleSpec/BrazeReactModuleSpec.h:183:22: error: redefinition of 'NativeBrazeReactModuleSpecJSI'
    class JSI_EXPORT NativeBrazeReactModuleSpecJSI : public ObjCTurboModule {
                     ^
In file included from /Users/expo/workingdir/build/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.mm:1:
In file included from /Users/expo/workingdir/build/node_modules/@braze/react-native-sdk/iOS/BrazeReactBridge/BrazeReactBridge/BrazeReactBridge.h:7:
/Users/expo/Library/Developer/Xcode/DerivedData/Yazen-bfgkgmgfisrfdgdyeaapyzbthuee/Build/Intermediates.noindex/ArchiveIntermediates/Yazen/BuildProductsPath/Release-iphoneos/React-Codegen/React_Codegen.framework/Headers/BrazeReactModuleSpec/BrazeReactModuleSpec.h:183:22: note: previous definition is here
    class JSI_EXPORT NativeBrazeReactModuleSpecJSI : public ObjCTurboModule {
 ```

### Verbose Logs

```shell
Might contain sensitive information. But I can email it if you want.
```

**Comment by jerielng** [comment]
Hey @ottob, thanks for raising this. It looks to be an issue with the way the native module spec is defined in our React Native SDK. We'll look into getting a fix for this and will update you when it's been released. Most likely, you will need to update the dependency for our [React Native SDK](https://github.com/braze-inc/braze-react-native-sdk) to that version rather than for our Expo plugin.

**Comment by ottob** [comment]
Thanks for the quick reply, looking forward to an update with the fix.

**Comment by jerielng** [comment]
Hey @ottob, this should now be resolved in version [8.4.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/8.4.0) of our React Native SDK. I'll close out this issue, but feel free to reach back out if you have further questions. Thank you!
